src/metabase/query_processor/middleware/process_userland_query.clj

```diff
@@ -170,14 +173,15 @@
       (let [query          (assoc-in query [:info :query-hash] (qp.util/query-hash query))
             execution-info (query-execution-info query)]
         (letfn [(rff* [metadata]
-                  (let [preprocessed-query (:preprocessed_query metadata)
+                  (let [#_preprocessed-query #_(:preprocessed_query metadata)
                         ;; we only need the preprocessed query to find field usages, so make sure we don't return it
                         result             (rff (dissoc metadata :preprocessed_query))
                         ;; skip internal queries as it use honeysql, not mbql
-                        field-usages       (when-not (qp.util/internal-query? query)
-                                             (field-usage/pmbql->field-usages
-                                              (lib/query (qp.store/metadata-provider) preprocessed-query)))]
-                    (add-and-save-execution-metadata-xform! execution-info field-usages result)))]
+                        ;; temporarily disabled because it impacts query performance
+                        #_field-usages       #_(when-not (qp.util/internal-query? query)
```

Comment:
we skip inserting field usage if it's nil in https://github.com/metabase/metabase/blob/disable-field-usage/src/metabase/query_processor/middleware/process_userland_query.clj#L53